lms/djangoapps/learner_dashboard/api/v0/views.py

```diff
@@ -355,15 +361,48 @@ class CourseRecommendationApiView(APIView):
     authentication_classes = (JwtAuthentication, SessionAuthenticationAllowInactiveUser,)
     permission_classes = (IsAuthenticated,)
 
+    def _emit_recommendations_viewed_event(
+        self, user_id, is_control, recommended_courses, amplitude_recommendations=True,
+    ):
+        """ Emits an event to track student dashboard page visits."""
+        segment.track(
+            user_id,
+            'edx.bi.user.recommendations.viewed',
+            {
+                'is_control': is_control,
+                'amplitude_recommendations': amplitude_recommendations,
+                'course_key_array': [course['course_key'] for course in recommended_courses],
+            }
+        )
+
+    def _general_recommendations_response(self, user_id, is_control, recommendations):
+        """ Helper method for general recommendations response """
+        self._emit_recommendations_viewed_event(user_id, is_control, recommendations, amplitude_recommendations=False)
+        return Response(
+            {
+                "courses": recommendations,
+                "is_control": is_control,
+            },
+            status=200,
+        )
+
     def get(self, request):
         """ Retrieves course recommendations details of a user in a specified course. """
         recommended_courses = []
         user_id = request.user.id
-        is_control, has_is_control, course_keys = get_personalized_course_recommendations(user_id)
+        fallback_recommendations = settings.GENERAL_RECOMMENDATIONS if show_fallback_recommendations() else []
 
-        if is_control or not course_keys:
-            self._emit_recommendations_viewed_event(user_id, is_control, has_is_control, recommended_courses)
-            return Response(status=400)
+        try:
+            is_control, has_is_control, course_keys = get_personalized_course_recommendations(user_id)
+        except Exception as ex:  # pylint: disable=broad-except
+            logger.warning(f"Cannot get recommendations from Amplitude: {ex}")
+            return self._general_recommendations_response(user_id, None, fallback_recommendations)
+
+        show_personalized_recommendations = not is_control if has_is_control else False
+        is_control = is_control if has_is_control else None
+
+        if not (show_personalized_recommendations and course_keys):
```

Comment:
```suggestion
       is_control = is_control if has_is_control else None
       if is_control or is_control is None or not course_keys:
```

nit: can we please check if we can simplify this condition?